app/src/main/java/com/douglasbruce/tasky/features/event/EventViewModel.kt

```diff
@@ -40,6 +42,77 @@ class EventViewModel @Inject constructor(
             is EventFormEvent.OnEditorSave -> {
                 state = state.copy(title = event.title, description = event.description)
             }
+
+            is EventFormEvent.OnHideTimePicker -> {
+                state = state.copy(showTimePicker = false)
+            }
+
+            is EventFormEvent.OnHideDatePicker -> {
+                state = state.copy(showDatePicker = false)
+            }
+
+            is EventFormEvent.OnTimePickerClick -> {
+                state = state.copy(
+                    isEditingToTime = event.isToTime,
+                    showTimePicker = true,
+                )
+            }
+
+            is EventFormEvent.OnTimeSelected -> {
+                val localTime: LocalTime = LocalTime.of(event.hour, event.minute)
+
+                state = if (state.isEditingToTime) {
+                    state.copy(
+                        toTime = localTime,
+                        showTimePicker = false,
+                    )
+                } else {
+                    val truncatedLocalTime = localTime.truncatedTo(ChronoUnit.MINUTES)
+                    val truncatedToTime = state.toTime.truncatedTo(ChronoUnit.MINUTES)
+
+                    val toTime = if (truncatedLocalTime == truncatedToTime || truncatedLocalTime.isAfter(truncatedToTime)) {
+                        localTime.plusMinutes(30)
+                    } else {
+                        state.toTime
+                    }
+
+                    state.copy(
+                        fromTime = localTime,
+                        toTime = toTime,
+                        showTimePicker = false,
+                    )
+                }
+            }
+
+            is EventFormEvent.OnDatePickerClick -> {
+                state = state.copy(
+                    isEditingToDate = event.isToDate,
+                    showDatePicker = true,
+                )
+            }
+
+            is EventFormEvent.OnDateSelected -> {
+                val localDate = DateUtils.getLocalDate(event.dateMillis)
+
+                state = if (state.isEditingToDate) {
+                    state.copy(
+                        toDate = localDate,
+                        showDatePicker = false,
+                    )
+                } else {
+                    val toDate = if (localDate.isAfter(state.toDate)) {
```

Comment:
Fun fact: If I remember correctly, you can also use normal comparison operators here: `if(localDate > state.toDate)`

Not that you have to, just in case you didn't know 😄